dependencyManagement/build.gradle.kts

```diff
@@ -0,0 +1,118 @@
+}
+
+data class DependencySet(val group: String, val version: String, val modules: List<String>)
+
+val dependencyVersions = hashMapOf<String, String>()
+rootProject.extra["versions"] = dependencyVersions
+
+val DEPENDENCY_BOMS = listOf(
+  "com.google.guava:guava-bom:30.1.1-jre",
+  "io.opentelemetry:opentelemetry-bom:1.2.0",
+  "io.opentelemetry:opentelemetry-bom-alpha:1.2.0-alpha",
+  "org.junit:junit-bom:5.7.1",
+)
+
+val DEPENDENCY_SETS = listOf(
+  DependencySet(
+    "com.google.auto.value",
+    "1.7.5",
+    listOf("auto-value", "auto-value-annotations")
+  ),
+  DependencySet(
+    "com.google.errorprone",
+    "2.4.0",
+    listOf("error_prone_annotations", "error_prone_core")
+  ),
+  DependencySet(
+    "io.prometheus",
+    "0.9.0",
+    listOf("simpleclient", "simpleclient_common", "simpleclient_httpserver")
+  ),
+  DependencySet(
+    "net.bytebuddy",
+    "1.10.18",
+    listOf("byte-buddy", "byte-buddy-agent")
+  ),
+  DependencySet(
+    "org.mockito",
+    "3.8.0",
+    listOf("mockito-core", "mockito-junit-jupiter")
+  ),
+  DependencySet(
+    "org.slf4j",
+    "1.7.30",
+    listOf("slf4j-simple", "log4j-over-slf4j", "jcl-over-slf4j", "jul-to-slf4j")
+  )
+  DependencySet(
+    "org.testcontainers",
+    "1.15.2",
+    listOf("testcontainers", "junit-jupiter")
+  )
+)
+
+val DEPENDENCIES = listOf(
+  "ch.qos.logback:logback-classic:1.2.3",
+  "com.blogspot.mydailyjava:weak-lock-free:0.18.0",
+  "com.github.ben-manes.caffeine:caffeine:2.9.0",
+  "com.github.stefanbirkner:system-lambda:1.1.0"
+  "com.google.auto.service:auto-service:1.0",
+  "com.uber.nullaway:nullaway:0.9.1",
+  "org.assertj:assertj-core:3.19.0",
+  "org.awaitility:awaitility:4.0.3",
+  "org.checkerframework:checker-qual:3.6.1",
+  "org.codehaus.groovy:groovy-all:2.5.11",
+  "org.objenesis:objenesis:3.1",
+  "org.spockframework:spock-core:1.3-groovy-2.5",
+  "org.scala-lang:scala-library:2.11.12",
+  "org.springframework.boot:spring-boot-dependencies:2.3.1.RELEASE"
+)
+
+javaPlatform {
+  allowDependencies()
+}
+
+dependencies {
+  for (bom in DEPENDENCY_BOMS) {
+    api(enforcedPlatform(bom))
+    val split = bom.split(':')
+    dependencyVersions[split[0]] = split[2]
+  }
+  constraints {
+    for (set in DEPENDENCY_SETS) {
+      for (module in set.modules) {
+        api("${set.group}:${module}:${set.version}")
+        dependencyVersions[set.group] = set.version
+      }
+    }
+    for (dependency in DEPENDENCIES) {
+      api(dependency)
+      val split = dependency.split(':')
+      dependencyVersions[split[0]] = split[2]
+    }
+  }
+}
+
+fun isNonStable(version: String): Boolean {
+  val stableKeyword = listOf("RELEASE", "FINAL", "GA").any { version.toUpperCase().contains(it) }
+  val regex = "^[0-9,.v-]+(-r)?$".toRegex()
+  val isGuava = version.endsWith("-jre")
+  val isStable = stableKeyword || regex.matches(version) || isGuava
+  return isStable.not()
+}
+
+tasks {
+  named<DependencyUpdatesTask>("dependencyUpdates") {
```

Comment:
Nice!

By the way, are we switching the instrumentation project gradle files to kotlin too?

Reply:
Doh! I accidentally added this into the wrong PR, still working on it :) I need to verify the impact of all of our projects and performance of kotlin but I am hoping to do so. But before that I want to make the dependency management consistent with SDK repo